### PR TITLE
ProjectionManager can now return ValueObject when asking for status

### PR DESCRIPTION
**Pull request by unixslayer**
resolves prooph/event-store#390

### CODE REVIEW

**Comment by enumag** [comment]
Same issue here ofc...

File: src/Projections/ProjectionsClient.php [hunk 1]
```diff
@@ -329,6 +331,27 @@ public function getStatus(
...
             HttpStatusCode::OK
         );
+
+        $promise->onResolve(function (?Throwable $exception, ?string $body) use ($deferred): void {
+            if ($exception) {
+                $deferred->fail($exception);
+
+                return;
+            }
+
```
Comment (prolic) [inline src/Projections/ProjectionsClient.php:341]:
```
if (null === $body) {
    $deferred->resolve(new UnexpectedValueException('No content received');

    return;
}

Comment (prolic) [reply src/Projections/ProjectionsClient.php:341]: I added the same [here](https://github.com/prooph/event-store-client/pull/137) for the other method

**Comment by prolic** [comment]
@unixslayer Tests are failing, see https://travis-ci.com/github/prooph/event-store-client/jobs/328969745

**Comment by unixslayer** [comment]
@prolic saw that, already pushed fixes

**Comment by prolic** [comment]
awesome, thank you !!

**Pull request merged**